mir_eval/transcription.py

```diff
@@ -0,0 +1,376 @@
+    2. The pitch of ref note i is within ``pitch_tolerance`` of the pitch of
+       est note j.
+    3. If ``offset_ratio`` is not ``None``, the offset of ref note i has to be
+       within ``offset_tolerance`` of the offset of est note j, where
+       ``offset_tolerance`` is equal to half the window given by taking
+       ``offset_ratio`` of the ref note's duration, i.e. ``0.5 * offset_ratio *
+       ref_duration[i]`` where ``ref_duration[i] = ref_intervals[i, 1] -
+       ref_intervals[i, 0]``. If the resulting ``offset_tolerance`` is less
+       than 0.05 (50 ms), 0.05 is used instead.
+    4. If ``offset_ratio`` is ``None``, note offsets are ignored, and only
+       criteria 1 and 2 are taken into consideration.
+
+    Every ref note is matched against at most one est note.
+
+    This is useful for computing precision/recall metrics for note
+    transcription.
+
+    Parameters
+    ----------
+    ref_intervals : np.ndarray, shape=(n,2)
+        Array of reference notes time intervals (onset and offset times)
+    ref_pitches: np.ndarray, shape=(n,)
+        Array of reference pitch values in Hertz
+    est_intervals : np.ndarray, shape=(m,2)
+        Array of estimated notes time intervals (onset and offset times)
+    est_pitches : np.ndarray, shape=(m,)
+        Array of estimated pitch values in Hertz
+    onset_tolerance : float > 0
+        The tolerance for an estimated note's onset deviating from the
+        reference note's onset, in seconds. Default is 0.05 (50 ms).
+    pitch_tolerance: float > 0
+        The tolerance for an estimated note's pitch deviating from the
+        reference note's pitch, in cents. Default is 50.0 (50 cents).
+    offset_ratio: float > 0 or None
+        The ratio of the reference note's duration used to define the
+        offset_tolerance. Default is 0.2 (20%), meaning the offset_tolerance
+        will equal the ref_duration * 0.2 * 0.5 (0.5 since the window is
+        centered on the reference offset), or 0.05 (50 ms), whichever is
+        greater. If ``offset_ratio`` is set to ``None``, offsets are ignored in
+        the matching.
+    offset_min_tolerance: float > 0
+        The minimum tolerance for offset matching. See offset_ratio description
+        for an explanation of how the offset tolerance is determined. Note:
+        this parameter only influences the results if ``offset_ratio`` is not
+        ``None``.
+    strict: bool
+        If ``strict=False`` (the default), threshold checks for onset, offset,
+        and pitch matching are performed using ``<=`` (less than or equal). If
+        ``strict=True``, the threshold checks are performed using ``<`` (less
+        than).
+
+    Returns
+    -------
+    matching : list of tuples
+        A list of matched reference and estimated notes.
+        ``matching[i] == (i, j)`` where reference note i matches estimate note
+        j.
+    """
+    # set the comparison function
+    if strict:
+        cmp = np.less
```

Comment:
`cmp` is a Python builtin: https://docs.python.org/2/library/functions.html#cmp :(